system/lib/pthread/emscripten_yield.c

```diff
@@ -7,10 +7,10 @@
 #include "syscall.h"
 #include "threading_internal.h"
 
-static _Atomic bool thread_crashed = false;
+static _Atomic pthread_t crashed_thread_id = NULL;
 
 void _emscripten_thread_crashed() {
-  thread_crashed = true;
+  crashed_thread_id = pthread_self();
```

Comment:
Can we not just call `_emscripten_thread_set_strongref(pthread_self())` here and avoid the change to `_emscripten_yield` below?

Reply:
I considered this, but then I noticed this comment:
https://github.com/emscripten-core/emscripten/blob/96d3452153f09a9f2d3e18695e9b956baf82464e/system/lib/pthread/emscripten_yield.c#L25-L29

So since this function is called from a crashed thread, I don't think we can use the normal proxying mechanism here (i.e. `__proxy: 'sync'`) .